api/lql.go

```diff
@@ -19,151 +19,44 @@
-		return nil
-	}
-	q.EndTimeRange = end
-
-	return nil
+type Query struct {
+	QueryID        string                   `json:"queryId"`
+	QueryText      string                   `json:"queryText"`
+	EvaluatorID    string                   `json:"evaluatorId"`
+	Owner          string                   `json:"owner"`
+	LastUpdateTime string                   `json:"lastUpdateTime"`
+	LastUpdateUser string                   `json:"lastUpdateUser"`
+	ResultSchema   []map[string]interface{} `json:"resultSchema"`
 }
 
-func (q *LQLQuery) TranslateQuery() error {
-	// if query text is already populated
-	if q.QueryText != "" {
-		return nil
-	}
-	// valid json
-	var t LQLQuery
-	if err := json.Unmarshal([]byte(q.QueryBlob), &t); err == nil {
-		if q.StartTimeRange == "" {
-			q.StartTimeRange = t.StartTimeRange
-		}
-		if q.EndTimeRange == "" {
-			q.EndTimeRange = t.EndTimeRange
-		}
-		q.QueryText = t.QueryText
-		return err
-	}
-	// invalid json
-	qblob := strings.ToLower(q.QueryBlob)
-	if strings.Contains(qblob, "start_time_range") ||
-		strings.Contains(qblob, "end_time_range") ||
-		strings.Contains(qblob, "lql_id") ||
-		strings.Contains(qblob, "query_text") {
-
-		return errors.New(LQLQueryTranslateError)
-	}
-	// valid lql text
-	if strings.Contains(q.QueryBlob, "{") &&
-		strings.Contains(q.QueryBlob, "}") {
-		q.QueryText = q.QueryBlob
-
-		return nil
-	}
-	// invalid lql text
-	return errors.New(LQLQueryTranslateError)
+type QueryResponse struct {
+	Data    Query  `json:"data"`
+	Message string `json:"message"`
 }
 
-func (q LQLQuery) TranslateTime(inTime string) (string, error) {
-	// empty
-	if inTime == "" {
-		return "", nil
-	}
-	// parse time as relative
-	if t, err := lwtime.ParseRelative(inTime); err == nil {
-		return t.UTC().Format(time.RFC3339), err
-	}
-	// parse time as RFC3339
-	if t, err := time.Parse(time.RFC3339, inTime); err == nil {
-		return t.UTC().Format(time.RFC3339), err
-	}
-	// parse time as millis
-	if i, err := strconv.ParseInt(inTime, 10, 64); err == nil {
-		return time.Unix(0, i*int64(time.Millisecond)).UTC().Format(time.RFC3339), err
-	}
-	return "", errors.New(fmt.Sprintf("unable to parse time (%s)", inTime))
+type QueriesResponse struct {
+	Data    []Query `json:"data"`
+	Message string  `json:"message"`
 }
 
-func (q LQLQuery) ValidateRange(allowEmptyTimes bool) (err error) {
-	// validate start
-	var start time.Time
-	if q.StartTimeRange != "" {
-		if start, err = time.Parse(time.RFC3339, q.StartTimeRange); err != nil {
-			return
-		}
-	} else if allowEmptyTimes {
-		start = time.Unix(0, 0)
-	} else {
-		err = errors.New("start time must not be empty")
-		return
-	}
-	// validate end
-	var end time.Time
-	if q.EndTimeRange != "" {
-		if end, err = time.Parse(time.RFC3339, q.EndTimeRange); err != nil {
-			return
-		}
-	} else if allowEmptyTimes {
-		end = time.Now()
-	} else {
-		err = errors.New("end time must not be empty")
-		return
-	}
+func validateQueryRange(start, end time.Time) (err error) {
```

Comment:
nit: curious if this time validation exists already inside `lwtime`, if not, disregard.

Reply:
We do not have validation in lwtime to ensure that a start time precedes end time, but this would be trivial to relocate to that package if so desired.